File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.java
class Solution {
    public int xorAllNums(int[] nums1, int[] nums2) {
        int n1 = nums1.length;
        int n2 = nums2.length;

        int ans = 0, x1 = 0, x2 = 0;

        for(int i = 0; i < n1; i++){
            x1^=nums1[i];
        }
        for(int i = 0; i < n2; i++){
            x2^=nums2[i];
        }

        if(n1 % 2 == 1){
            ans^=x2;
        }

        if(n2 % 2 == 1){
            ans^=x1;
        }


        return ans;
    }
}

File: 2533-bitwise-xor-of-all-pairings/2533-bitwise-xor-of-all-pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();

        int res = 0, x1 = 0, x2 = 0;
        for(int n1 : nums1){
            x1^= n1;
        }

        for(int n2 : nums2){
            x2^=n2;
        }

        if(n%2 == 1){
            res^= x2;
        }

        if(m%2 == 1){
            res^=x1;
        }

        return res;
    }
};
/*

[2,1,3]
[10,2,5,0]




*/